src/miral/basic_window_manager.h

```diff
@@ -320,6 +324,9 @@ class BasicWindowManager : public virtual mir::shell::WindowManager,
     void advise_output_end() override;
     /// Updates the application zones of all display areas and moves attached windows as needed
     void update_application_zones_and_attached_windows();
+
+    /// Collects all children windows provided a WindowInfo
+    auto collect_windows(WindowInfo const& info) -> std::vector<Window>;
```

Comment:
Everywhere it is used the vector returned by `collect_windows()` is copied into a `SurfaceSet`, why doesn't it return a `SurfaceSet` and avoid all that copying?